controllers/storagecluster/provider_server.go

```diff
@@ -63,6 +67,30 @@ func (o *ocsProviderServer) ensureCreated(r *StorageClusterReconciler, instance
 		return res, nil
 	}
 
+	if os.Getenv(onboardingSecretGeneratorImage) == "" {
+		r.Log.Info("onboardingSecretGeneratorImage env var is not set.")
+	} else {
+		onboardingSecretGeneratorJob := &batchv1.Job{}
+
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: instance.Namespace, Name: onboardingJobName}, onboardingSecretGeneratorJob)
+
+		if err != nil && errors.IsAlreadyExists(err) {
+			for _, condition := range onboardingSecretGeneratorJob.Status.Conditions {
+				if condition.Type == batchv1.JobComplete {
+					r.Log.Info("Job already exists and in completed status")
+				}
+			}
+
+		} else if err != nil && errors.IsNotFound(err) {
+			onboardingSecretGeneratorJob = createOnboardingJob(instance)
+			err = r.Client.Create(context.Background(), onboardingSecretGeneratorJob)
+			if err != nil {
+				r.Log.Error(err, "Failed to create job.")
```

Comment:
we want to create a job isn't it? we should return error and ask for another reconile

Reply:
true, did the change